tern/formats/cyclonedx/cyclonedx_common.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+"""
+Common functions that are useful for CycloneDX document creation
+"""
+
+import datetime
+import uuid
+from tern.utils import general
+
+
+###################
+# General Helpers #
+###################
+
+
+# document level tool information
+metadata_tool = {
+    'vendor': 'Tern Tools',
+    'name': 'Tern',
+    'version': general.get_git_rev_or_version()[1]
+}
+
+
+# keys are what Tern uses, values what CycloneDX uses
+hash_type_mapping = {
+    'md5': 'MD5',
+    'sha1': 'SHA-1',
+    'sha256': 'SHA-256',
+}
+
+
+pkg_format_purl_type_mapping = {
+    'deb': 'deb',
+    'rpm': 'rpm',
+    'apk': 'apk',
+    'pip': 'pip',
+    'gem': 'gem',
+    'npm': 'npm',
+    'go.mod': 'go',
```

Comment:
If `go.mod` is the only one that is different for purl, you can change the `pkg_format` in `tern/analyze/default/command_lib/base.yml`:
```
go:
  pkg_format: 'go.mod' <-- change this to 'go'
```

Then you will not need this mapping.